core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -95,6 +95,10 @@ public static PluginInfo readFromProperties(Path dir) throws IOException {
         if (version == null) {
             throw new IllegalArgumentException("Property [version] is missing for plugin [" + name + "]");
         }
+        String pluginName = props.getProperty("plugin.name");
+        if (pluginName == null) {
+            throw new IllegalArgumentException("Property [plugin.name] is missing for plugin [" + name + "]");
+        }
```

Comment:
I think the whole point of this PR is to add this new parameter and make it required, so I don't think we should fall back? Otherwise the leniency that we are trying to fix is still here.